internal/client/job.go

```diff
@@ -325,6 +328,40 @@ func (c *Project) queueAndStreamJob(
 				stateEventTimer = nil
 			}
 
+			// Check if this job has been assigned a runner for the first time
+			if event.State != nil &&
+				event.State.Job != nil &&
+				event.State.Job.AssignedRunner != nil &&
+				assignedRunner == nil {
+
+				assignedRunner = event.State.Job.AssignedRunner
+
+				runner, err := c.client.GetRunner(ctx, &pb.GetRunnerRequest{RunnerId: assignedRunner.Id})
+				if err != nil {
+					c.UI.Output("Failed to inspect the runner (id %q) assigned for this operation: %s", assignedRunner.Id, err, terminal.WithErrorStyle())
+				}
+				switch runnerType := runner.Kind.(type) {
+				case *pb.Runner_Local_:
+					c.UI.Output("Performing operation locally", terminal.WithInfoStyle())
```

Comment:
All these should use `ui` not `c.UI`. I don't remember why, but we should be using the passed in UI that is identical to the one we use for other output, shouldn't we?